<?xml version="1.0"?>
<!--
Launch the path follower with the PID controller for a car-like (Ackermann) robot.
-->
<launch>
    <node pkg="path_follower" type="path_follower_node" name="path_follower" output="screen" respawn="true">
        <param name="controller_type" value="ackermann_pid" />
    </node>

    <group ns="path_follower">
        <param name="min_velocity" value="0.5" />
        <param name="max_velocity" value="1.0" />
        <param name="waypoint_tolerance" value="0.8" />
        <param name="goal_tolerance" value="0.4" />

        <group ns="controller">
            <!-- parameters, that are dependent of the controller or the robot model -->
            <group ns="ackermann_pid">
                <param name="pid/kp" value="1.5" />
                <param name="pid/ki" value="0.001" />
            </group>
        </group>

        <group ns="collision_avoider">
            <group ns="collision_box">
                <param name="width" value="0.4" />
                <param name="crit_length" value="0.5" />
                <param name="min_length" value="0.7" />
                <param name="max_length" value="1.4" />
                <param name="velocity_factor" value="1.0" />
                <param name="velocity_saturation" value="1.0" />
            </group>
        </group>

        <group ns="supervisor">
            <param name="use_waypoint_timeout" value="true" />
            <group ns="waypoint_timeout">
                <param name="time" value="30.0" />
            </group>

            <param name="use_distance_to_path" value="true" />
            <group ns="distance_to_path">
                <param name="max_dist" value="2.0" />
            </group>

            <param name="use_path_lookout" value="true" />
            <group ns="path_lookout">
                <param name="obstacle_scale_distance" value="1.0" />
                <param name="obstacle_scale_lifetime" value="5.0" />
                <param name="path_width" value="0.5" />
            </group>
        </group>

    </group>
  
</launch>
